org.eclipse.sprotty.layout/src/main/java/org/eclipse/sprotty/layout/ElktSerializer.java

```diff
@@ -0,0 +1,77 @@
+/********************************************************************************
+ * Copyright (c) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.sprotty.layout;
+
+import java.io.ByteArrayOutputStream;
+import java.lang.reflect.Method;
+import java.nio.charset.Charset;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import org.apache.log4j.Logger;
+import org.eclipse.elk.graph.ElkNode;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+
+
+/**
+ * Allows to dump an ELK model in ELKT, such that it can be debugged in 
+ * https://rtsys.informatik.uni-kiel.de/elklive/elkgraph.html 
+ * 
+ * This class needs the Maven artifact
+ *   groupID: org.eclipse.elk
+ *   artifactId: org.eclipse.elk.graph.text
+ * on the classpath.
+ */
+public class ElktSerializer {
+		
+	private static final Logger LOG = Logger.getLogger(ElktSerializer.class);
+	
+	@Inject Provider<ResourceSetImpl> resourceSetProvider;
```

Comment:
Shall we add a setter to allow setting up an instance without DI? Then `@Inject` can move to the setter.